scripts/composer/ScriptHandler.php

```diff
@@ -88,6 +88,77 @@ public static function downloadDevelopPackages(Event $event) {
     }
   }
 
+  /**
+   * Reset local repositories to the default branch.
+   *
+   * @param \Composer\Script\Event $event
+   *   The script event.
+   */
+  public static function resetLocalRepositories(Event $event) {
+    $io = $event->getIO();
+    $repositoriesInfo = self::getLocalRepositoriesInfo($event);
+    $drupalFinder = new DrupalFinder();
+    $drupalFinder->locateRoot(getcwd());
+    $composerRoot = $drupalFinder->getComposerRoot();
+
+    $io->write(PHP_EOL);
+    foreach ($repositoriesInfo as $key => $info) {
+      $localBranch = trim(shell_exec('git -C ' . $composerRoot . '/' . $info['install_path'] . ' rev-parse --abbrev-ref HEAD'));
```

Comment:
Maybe it would be good to create a variable with git command for a path - `$git_with_path = 'git -C ' . $composerRoot . '/' . $info['install_path']` - then you can use that everywhere and the code would be a bit cleaner and more readable.